### DISCUSSION TITLE
Custumize E-Mail text without use of template files

### DISCUSSION THREAD

**Discussion by capoaira** [iMerica/dj-rest-auth#636] [Q&A]
Hello,
I want to customize the E-Mail text for `/api/rest-auth/password/reset/` as well as `/api/rest-auth/registration/` and `/api/rest-auth/registration/resend-email/` (resend-email can have the same text then registration)

The best way may be a custom serializer, but I wasn't able to create one, which generate a valid link (for password reset as well as email verification).

These are my main problems:
- The link has to fit to the environment
- I need the correct token
- the confirm pages are self written with vuejs
- The text should be loaded from a db (I know how, but not where to do this)

I already tried a lot, but without success. Can somebody help me?

**Comment by ra-dave** [comment]
@capoaira  Did you ever find a solution?  I am trying the custom serializer approach but it never gets called.

**Reply by capoaira** [reply]
Yes, but it was a long and ugly way, and different emails has to be overridden on different places in multiple files. I'm not sure if I can find everything again 😅. But if you have a question to one specific email override, I'll try to reconstruct my steps :)

To get your custom serializer get called, you have to override `REST_AUTH` in your settings, e.g.
```py
REST_AUTH = {
    'USER_DETAILS_SERIALIZER': 'your_project.users.api.serializers.UserSerializer',
    'REGISTER_SERIALIZER': 'your_project.users.api.serializers.RegisterSerializer',
    # ... your other REST_AUTH setiings
}
```

What me helped a lot, was trying to understand how these emails were sent by the framework, so I download the source code of dj-rest-auth and all auth. (I did not understand everything, but enough to override the emails).
